# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1212639784 14400
#      Thu Jun 05 00:23:04 2008 -0400
# Node ID 1b85fcb94c6a42aae2e2291d3948812b09691e0c
# Parent  f6fffa74b9b5f3d00eaaa8e3730697cb0c6e165d
properties::update_normals: eliminate unnecessary use of references

diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,8 +1,13 @@
+2008-06-05  John W. Eaton  <jwe@octave.org>
+
+	* graphics.cc (properties::update_normals):
+	Eliminate unnecessary use of references.
+
 2008-06-04  Shai Ayal  <shaiay@users.sourceforge.net>
 
 	* graphics.cc (axes::properties::set_defaults): Preserve position
 	if mode is replace.
 	(axes::properties::sync_positions): Use default_axes_postion for
 	consistency.
 
 	* graphics.h.in (axes::properties::sync_positions,
diff --git a/src/graphics.cc b/src/graphics.cc
--- a/src/graphics.cc
+++ b/src/graphics.cc
@@ -3444,19 +3444,19 @@ surface::properties::update_normals (voi
 	    {
 	      if (x_mat)
 		{
 		  j1 = j-1;
 		  j2 = j;
 		  j3 = j+1;
 		}
 
-	      double& nx = n(j, i, 0);
-	      double& ny = n(j, i, 1);
-	      double& nz = n(j, i, 2);
+	      double nx = n(j,i,0);
+	      double ny = n(j,i,1);
+	      double nz = n(j,i,2);
 
 	      cross_product (x(j3,i)-x(j2,i), y(j+1,i2)-y(j,i2), z(j+1,i)-z(j,i),
 			     x(j2,i+1)-x(j2,i), y(j,i3)-y(j,i2), z(j,i+1)-z(i,j),
 			     nx, ny, nz);
 	      cross_product (x(j2,i-1)-x(j2,i), y(j,i1)-y(j,i2), z(j,i-1)-z(j,i),
 			     x(j3,i)-x(j2,i), y(j+1,i2)-y(j,i2), z(j+1,i)-z(i,j),
 			     nx, ny, nz);
 	      cross_product (x(j1,i)-x(j2,i), y(j-1,i2)-y(j,i2), z(j-1,i)-z(j,i),
